libs/cypress-indexeddb/src/lib/cypress-indexeddb.ts

```diff
@@ -0,0 +1,167 @@
+const DATABASES = new Map<string, IDBDatabase>();
+const STORES = new Map<string, IDBObjectStore>();
+
+export function setupIDBHelpers(): void {
+  // -- This is a parent command --
+  Cypress.Commands.add('clearIndexedDb', (databaseName: string) => {
+    return cy.window().then(async (window: Cypress.AUTWindow) => {
+      const promise = new Promise<void>((resolve, reject) => {
+        const deleteDb = window.indexedDB.deleteDatabase(databaseName);
+        deleteDb.onsuccess = () => {
+          if (DATABASES.has(databaseName)) {
+            DATABASES.delete(databaseName);
+          }
+          resolve();
+        };
+        deleteDb.onerror = (e) => reject(e);
+      });
+      try {
+        await promise;
+        cy.log(`"${databaseName}" cleared`);
+      } catch (e) {
+        cy.log(`error during "${databaseName}" clear operation`);
+        cy.log(e);
+      }
+      return window;
+    });
+  });
+
+  Cypress.Commands.add(
+    'openIndexedDb',
+    (databaseName: string, versionConfiguredByUser?: number) => {
+      return cy.window().then(async (window: Cypress.AUTWindow) => {
+        const db = DATABASES.get(databaseName);
+        if (db) {
+          return Promise.resolve(db);
+        }
+        const newVersion = versionConfiguredByUser || 2;
+        const request = window.indexedDB.open(databaseName, newVersion);
+        return new Promise<IDBDatabase>((resolve, reject) => {
+          request.onerror = reject;
+          request.onupgradeneeded = (event: any) => {
```

Comment:
does it have to be typed as `any`? can we use `IDBVersionChangeEvent`
BTW, I've looked at TS type for this and it seems that the first param should be `this: IDBOpenDBRequest`

```ts
onupgradeneeded: ((this: IDBOpenDBRequest, ev: IDBVersionChangeEvent) => any) | null;
```

Reply:
I don't know why that is the case, but I am pretty sure it returns an event based on the MDN documentation.